targets/targets.json

```diff
@@ -3575,7 +3575,7 @@
             }
         },
         "inherits": ["Target"],
-        "macros": ["CMSIS_VECTAB_VIRTUAL", "CMSIS_VECTAB_VIRTUAL_HEADER_FILE=\"cmsis_nvic.h\""],
```

Comment:
We already enable other features like MBED_HEAP_STATS_ENABLED, MBED_STACK_STATS_ENABLED, MBED_TRAP_ERRORS_ENABLED on this platform for testing. So disabling mbed-os hardfault handler feature as this target is limited on ram/rom.